src/Microsoft.Extensions.FileProviders.Physical/Internal/PhysicalDirectoryContents.cs

```diff
@@ -17,19 +18,25 @@ public class PhysicalDirectoryContents : IDirectoryContents
     {
         private IEnumerable<IFileInfo> _entries;
         private readonly string _directory;
+        private readonly ExcludedFiles _excludedFiles;
 
         /// <summary>
         /// Initializes an instance of <see cref="PhysicalDirectoryContents"/>
         /// </summary>
         /// <param name="directory">The directory</param>
         public PhysicalDirectoryContents(string directory)
-        {
-            if (directory == null)
-            {
-                throw new ArgumentNullException(nameof(directory));
-            }
+            : this(directory, ExcludedFiles.Sensitive)
+        { }
 
-            _directory = directory;
+        /// <summary>
+        /// Initializes an instance of <see cref="PhysicalDirectoryContents"/>
+        /// </summary>
+        /// <param name="directory">The directory</param>
+        /// <param name="excludedFiles">Specifies which files are excluded from the directory listing.</param>
+        public PhysicalDirectoryContents(string directory, ExcludedFiles excludedFiles)
+        {
+            _directory = directory ?? throw new ArgumentNullException(nameof(directory));
```

Comment:
They have Fowler's (relucant) approval. I'll conceded throw expressions are not always the right choice, but I think they're a win for simple ctor checks.